opaque_keys/__init__.py

```diff
@@ -362,11 +392,13 @@ def __getstate__(self):
     @property
     def _key(self):
         """Returns a tuple of key fields"""
+        return unicode(self)
         # pylint: disable=no-member
-        return tuple(getattr(self, field) for field in self.KEY_FIELDS) + (self.CANONICAL_NAMESPACE, self.deprecated)
+        # return tuple(getattr(self, field) for field in self.KEY_FIELDS) + (self.CANONICAL_NAMESPACE, self.deprecated)
 
     def __eq__(self, other):
-        return isinstance(other, OpaqueKey) and self._key == other._key  # pylint: disable=protected-access
+        return unicode(self) == unicode(other)
```

Comment:
Why not use `return hash(self) == hash(other)` here? This code still does string comparison.